src/providers/trusted_service/context/key_management.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::ts_binding::{rpc_caller, rpc_caller_begin, rpc_caller_end, rpc_caller_invoke};
+use super::ts_protobuf::{
+    CloseKeyIn, DestroyKeyIn, DestroyKeyOut, GenerateKeyIn, GenerateKeyOut, KeyAttributes,
+    KeyLifetime, KeyPolicy, Opcode, OpenKeyIn, OpenKeyOut,
+};
+use super::Context;
+use log::{error, info};
+use parsec_interface::operations::psa_key_attributes::Attributes;
+use parsec_interface::requests::ResponseStatus;
+use prost::Message;
+use psa_crypto::types::status::{Error, Status};
+use std::convert::{TryFrom, TryInto};
+use std::mem::forget;
+use std::ptr::null_mut;
+
+impl Context {
+    pub fn generate_key(
+        &self,
+        key_attrs: Attributes,
+        id: u32,
+    ) -> parsec_interface::requests::Result<u32> {
+        let proto_req = Box::new(GenerateKeyIn {
+            attributes: Some(KeyAttributes {
+                r#type: u16::try_from(key_attrs.key_type)? as u32,
```

Comment:
Can't you directly do `u32::try_from`?

Reply:
I don't think so - `key_type` is an enum, and we've implemented `TryFrom<Type> for psa_crypto_sys::psa_key_type_t`. I then have to go from `psa_key_type_t` to `u32`.

Reply:
ahh I see! I did not even look but we are going from a small to big type so it's all good not using `try_from`.